File: Corte1/Tarea 120826Entrega140826/lectorpalindromo.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

    string numeroCodigo = "0";
    cout << "Este programa determina si un numero es un palindromo o no. \nIngrese un numero: ";
    cin >> numeroCodigo;
    // Se crea un array del tamaño de numeros y se rellena con los digtos usando un for
    char digitosNumero[numeroCodigo.length()];

    for (int i = 0; i < numeroCodigo.length(); i++ && numeroCodigo.length() > 0) {
        digitosNumero[i] = numeroCodigo[i];
    }

    for (int i= 0, j = numeroCodigo.length() - 1; i < j; i++, j--) {
        if (digitosNumero[i] != digitosNumero[j]) {
            cout << "El numero " << numeroCodigo << " no es un palindromo y por ende el codigo no es valido." << endl;
            return 0;
        }
    }
    cout << "El numero " << numeroCodigo << " es un palindromo y por ende el codigo es valido." << endl;
}

File: Corte1/Tarea 120826Entrega140826/lectorpalindromo.py
numeroCodigo = input("Este programa determina si un numero es un palindromo o no. \n\nIngrese un numero: ")
numeroString = str(numeroCodigo)
digitosNumero = list(numeroString)

for i in range(len(digitosNumero) // 2 and len(digitosNumero) != 0):
    #El -i + 1 es porque en python el negativo es el negativo del indice
    if digitosNumero[i] != digitosNumero[-(i + 1)]:
        print(f"El numero {numeroCodigo} no es un palindromo, es decir no es un codigo valido.")
        break
    else:
        print(f"El numero {numeroCodigo} es un palindromo, es decir el codigo es valido.")
